File: 392-is-subsequence/392-is-subsequence.py
class Solution:
    # O(n) time | O(1) space
    def isSubsequence(self, s: str, t: str) -> bool:
        subseqIndex = 0
        for letter in t:
            if subseqIndex == len(s): break
            if s[subseqIndex] == letter: subseqIndex += 1
        return subseqIndex == len(s)

File: 392-is-subsequence/392-is-subsequence.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
// O(n) time | O(1) space
var isSubsequence = function(s, t) {
    let subseqIndex = 0;
    for (let letter of t){
        if (subseqIndex === s.length) break;
        if (s[subseqIndex] === letter) subseqIndex++;
    }
    return subseqIndex === s.length;
};